yt_dlp/extractor/litv.py

```diff
@@ -68,73 +70,63 @@ def _real_extract(self, url):
+            default={}, expected_type=dict)
+        series_id = program_info.get('series_id')
+        playlist_data = traverse_obj(vod_data, ('props', 'pageProps', 'seriesTree'))
+        if playlist_data is not None and self._yes_playlist(series_id, video_id, smuggled_data):
+            content_type = program_info.get('content_type')
+            return self._extract_playlist(playlist_data, content_type)
+
+        asset_id = traverse_obj(program_info, ('assets', 0, 'asset_id'))
+        if asset_id is None:  # live stream case
+            asset_id = program_info.get('content_id')
+            media_type = program_info.get('content_type')
+        else:  # vod case
+            media_type = 'vod'
+        puid = try_call(lambda: self._get_cookies('https://www.litv.tv/')['PUID'].value)
+        if puid is None:
+            puid = str(uuid.uuid4())
+            endpoint = 'get-urls-no-auth'
+        else:
+            endpoint = 'get-urls'
+        payload = {'AssetId': asset_id, 'MediaType': media_type, 'puid': puid}
+        video_data = self._download_json(
+            f'https://www.litv.tv/api/{endpoint}', video_id,
+            data=json.dumps(payload).encode(),
+            headers={'Content-Type': 'application/json'})
+
+        if video_data.get('error'):
+            error_msg = traverse_obj(video_data, ('error', 'message'))
+            if error_msg == 'OutsideRegionError: ':
                 self.raise_geo_restricted('This video is available in Taiwan only')
             if error_msg:
                 raise ExtractorError(f'{self.IE_NAME} said: {error_msg}', expected=True)
-            raise ExtractorError(f'Unexpected result from {self.IE_NAME}')
+            raise ExtractorError(f'Unexpected error from {self.IE_NAME}')
 
+        try:
+            asset_url = video_data['result']['AssetURLs'][0]
+        except KeyError:
+            raise ExtractorError(f'Unexpected result from {self.IE_NAME}')
         formats = self._extract_m3u8_formats(
-            video_data['fullpath'], video_id, ext='mp4',
+            asset_url, video_id, ext='mp4',
             entry_protocol='m3u8_native', m3u8_id='hls')
         for a_format in formats:
             # LiTV HLS segments doesn't like compressions
             a_format.setdefault('http_headers', {})['Accept-Encoding'] = 'identity'
 
-        title = program_info['title'] + program_info.get('secondaryMark', '')
+        title = program_info['title'] + program_info.get('secondary_mark', '')
         description = program_info.get('description')
-        thumbnail = program_info.get('imageFile')
-        categories = [item['name'] for item in program_info.get('category', [])]
+        thumbnail = program_info.get('picture')
+        if thumbnail is not None:
+            thumbnail = 'https://p-cdnstatic.svc.litv.tv/' + thumbnail
+        categories = [item['name'] for item in program_info.get('genres', [])]
         episode = int_or_none(program_info.get('episode'))
```

Comment:
I would inline all this below in the return

import `functools`, and import `join_nonempty` and `urljoin` from `..utils`
```py
        return {
            'id': video_id,
            'formats': formats,
            'title': join_nonempty('title', 'secondary_mark', delim='', from_dict=program_info),
            **traverse_obj(program_info, {
                'description': ('description', {str}),
                'thumbnail': ('picture', {functools.partial(urljoin, 'https://p-cdnstatic.svc.litv.tv/')}),
                'categories': ('genres', ..., 'name', {str}),
                'episode_number': ('episode', {int_or_none}),
            }),
        }
```